pyodide_build/recipe/builder.py

```diff
@@ -329,11 +346,31 @@ def _download_and_extract(self) -> None:
         # is too large for the chown() call. This behavior can lead to "Permission denied" errors
         # (missing x bit) or random strange `make` behavior (due to wrong mtime order) in the CI
         # pipeline.
-        shutil.unpack_archive(
-            tarballpath,
-            self.build_dir,
-            filter=None if tarballpath.suffix == ".zip" else "data",
-        )
+        if tarballpath.suffix == ".zip":
+            shutil.unpack_archive(tarballpath, self.build_dir, filter=None)
+        else:
+            filetime = (
+                _get_source_epoch() if "SOURCE_DATE_EPOCH" in os.environ else None
+            )
+
+            def reproducible_filter(
+                tarinfo: tarfile.TarInfo, path: str | Path | None = None
+            ) -> tarfile.TarInfo:
```

Comment:
I'm not a fan of this method being nested in another method. However, there are similar patterns in this file as well.

Reply:
Thanks, addressed in 4c55ea8d5816cbf7336b1f5b210a7a0d6b074d02 by moving to a private non-nested method.